### PR TITLE
add special fn case to signature_to_binary

### LINKED ISSUE #8428: IEx doc header broken for `fn/1`
### Environment

* Elixir & Erlang/OTP versions (elixir --version): master
* Operating system: macOS Mojave

### Current behavior

screenshot 2018-11-26 at 12 04 21

### PR DESCRIPTION
**Pull request by Hanspagh**
Handles the special fn case in the signature_to_binary function. As of now I hardcoded the signature of fn, I am not sure this is the intended behavior? 
Fixes #8428

### CODE REVIEW

File: lib/elixir/test/elixir/code_test.exs [hunk 1]
```diff
@@ -293,6 +293,14 @@ defmodule CodeTest do
       Code.compiler_options(debug_info: :not_a_boolean)
     end
   end
+
+  describe "fecth docs handles special cases correct" do
```
Comment (fertapric) [inline lib/elixir/test/elixir/code_test.exs]: There is a [`docs_test.exs` file](https://github.com/elixir-lang/elixir/blob/master/lib/elixir/test/elixir/kernel/docs_test.exs), I would put this test there :)

File: lib/elixir/src/elixir_erl.erl [hunk 2]
```diff
@@ -531,6 +531,9 @@ get_type_docs(Set, Types) ->
 signature_to_binary(_Module, Name, _Signature) when Name == '__aliases__'; Name == '__block__' ->
   <<(atom_to_binary(Name, utf8))/binary, "(args)">>;
 
+signature_to_binary(_Module, Name, _Signature) when Name == fn -> 
+  <<"fn [clauses] end">>;
```
Comment (josevalim) [inline lib/elixir/src/elixir_erl.erl]: I think we can do `fn (clauses) end` or `fn ... end` or simply `fn`.

File: lib/elixir/test/elixir/code_test.exs (hunk 1 above)
Comment (Hanspagh) [reply lib/elixir/test/elixir/code_test.exs]: Makes sense, don't know why I didn't see that file before.

File: lib/elixir/src/elixir_erl.erl (hunk 2 above)
Comment (Hanspagh) [reply lib/elixir/src/elixir_erl.erl]: Would like make sense to do something like this `atom_to_binary(Name, utf8);` then?

Comment (josevalim) [reply lib/elixir/src/elixir_erl.erl]: In this case we know that the Name is atom, so we can skip the indirection. So we should rather write it like this:

```erlang
signature_to_binary(_Module, fn, _Signature) -> 
  <<"fn">>;
```

Comment (Hanspagh) [reply lib/elixir/src/elixir_erl.erl]: Fixed

File: lib/elixir/test/elixir/kernel/docs_test.exs [hunk 3]
```diff
@@ -307,4 +307,12 @@ defmodule Kernel.DocsTest do
              {{:fuz, 0}, :none}
            ] = Enum.sort(function_docs)
   end
+
+  describe "fecth docs handles special cases correct" do
```
Comment (maennchen) [inline lib/elixir/test/elixir/kernel/docs_test.exs]:
```suggestion
  describe "fetch docs handles special cases correct" do
```

Comment (fertapric) [reply lib/elixir/test/elixir/kernel/docs_test.exs]: Maybe?
 
```elixir
describe "special signatures" do
  test "fn" do
    ...
  end
end
```

**Pull request merged**